### ISSUE TITLE
Extremely slow start of epochs (running in Windows)

### ISSUE THREAD

**Issue by PinkysBrain** [kentaroy47/vision-transformers-cifar10#38]
I am trying to run this on windows with python 3.13 and Pytorch 2.6. To use this project in windows, the code in train_cifar10.py needed to use a check for main due to lack of fork. Also changing the code in utils to use "shutil.get_terminal_size()" instead of "os.popen('stty size', 'r').read().split()" prevents a lot of error spam.

At that point it runs, and when I try to train the mlpmixer I can enjoy extreme slowness of dataloader. 

"Epoch: 0
C:\Users\pinky\vision-transformers-cifar10-0.4\train_cifar10.py:42: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast(enabled=use_amp):
 [>................................................................]  Step: 42s99ms | Tot: 4ms | Loss: 2.366 | Acc: 10.1
 [>................................................................]  Step: 217ms | Tot: 221ms | Loss: 2.547 | Acc: 10.0
 [=>...............................................................]  Step: 211ms | Tot: 433ms | Loss: 2.560 | Acc: 11.5 
 ..."

Happens for every Epoch and also for testing. During that 42s it's just stuck on the CPU, I can see the upload of data happening in the resource monitor and it's almost instant (also it somehow fills 3.5 GB of memory, that seems too much even if the model is instantiated 8x). Is this a pytorch snafu or is it because I'm using windows?

I see a bug for pytorch, but if this is how it always works that seems almost unacceptable for actual use.
https://github.com/pytorch/pytorch/issues/106704
